kedro-airflow/kedro_airflow/grouping.py

```diff
@@ -113,3 +113,37 @@ def dfs(cur_node_name: str, component: int) -> None:
                 group_dependencies[new_name_parent].append(new_name_child)
 
     return group_to_seq, group_dependencies
+
+
+def group_by_namespace(
+    pipeline: Pipeline,
+) -> tuple[dict[str, list[Node]], dict[str, list[str]]]:
+    """
+    Groups nodes based on their namespace using Pipeline's grouped_nodes_by_namespace property.
+    Non-namespaced nodes are assigned to a default namespace.
+    """
+    nodes_by_namespace: dict[str, list[Node]] = {}
+    dependencies_by_namespace: dict[str, list[str]] = {}
+
+    grouped_nodes = pipeline.grouped_nodes_by_namespace
+
+    node_to_namespace = {}
+
+    for group_name, group_info in grouped_nodes.items():
+        ns = group_info["name"] if group_info["type"] == "namespace" else "__default__"
+        nodes_by_namespace.setdefault(ns, [])
+        dependencies_by_namespace.setdefault(ns, [])
+        nodes_by_namespace[ns].extend(group_info["nodes"])
+        for node in group_info["nodes"]:
+            node_to_namespace[node] = ns
+
+    for ns, nodes in nodes_by_namespace.items():
+        dependent_namespaces = set()
+        for node in nodes:
+            for parent in pipeline.node_dependencies.get(node, []):
+                parent_ns = node_to_namespace.get(parent, "__default__")
+                if parent_ns != ns:
+                    dependent_namespaces.add(parent_ns)
+        dependencies_by_namespace[ns] = list(dependent_namespaces)
```

Comment:
Why do you need to do this? `pipeline.grouped_nodes_by_namespace` also returns the dependencies for each namespace right? https://github.com/kedro-org/kedro/blob/main/kedro/pipeline/pipeline.py#L398